src/services/PaymentService.ts

```diff
@@ -346,17 +346,19 @@ export class PaymentService {
   }
 
   async getPrices(currency?: string): Promise<DisplayPrice[]> {
+    const currencyValue = currency ?? 'eur';
     const res = await this.provider.prices.search({
-      query: `metadata["show"]:"1" active:"true" currency:"${currency ?? 'eur'}"`,
+      query: `metadata["show"]:"1" active:"true" currency:"${currencyValue}"`,
+      expand: ['data.currency_options'],
       limit: 100,
     });
 
     return res.data
       .filter((price) => price.metadata.maxSpaceBytes)
       .map((price) => ({
         id: price.id,
-        currency: price.currency,
-        amount: price.unit_amount!,
+        currency: currencyValue,
```

Comment:
This could have unexpected behaviors, at least assuming the below code makes sense. 

Just below you are doing: 
`amount: price.currency_options?.[currencyValue].unit_amount ?? price.unit_amount!`

I do not know in which case, the unit_amount is not set on the currency_options, but let's assume is like that (in which case, that value should be properly set from Stripe, instead of us patching it with a fallback). You could have the amount in $ (`price.unit_amount`) and the currency in € (`currencyValue` default value). 

I would enforce the `currency` value to match the `price.currency` to avoid this situation. I would remove that price from the listing and add a log for the situation so that in case we add a new price and the currency is not properly set, we instantly realize of that.